Armbian-install: add option to wipe target destination

igorpecovnik opened the pull request:
# Description

armbian-installer have troubles if we are installing OS to SSD with previously defined partitions. Adding a recommended option to start with a clean drive - wiping everything - is one way. In this case, our installer starts in predicted way. Also it gives user no option to choose wrong (too small) partition, if there are more of them pre-made.

[Jira](https://armbian.atlassian.net/jira) reference number [AR-2393]
https://forum.armbian.com/topic/41699-how-to-debugfix-armbian-config-for-kernel-update/#comment-195551

# How Has This Been Tested?

- [x] Made several installs to nvme with several or no partitions

# Checklist:

- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules


[AR-2393]: https://armbian.atlassian.net/browse/AR-2393?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

hi-ko commented:
Although partitioning is not my issue, I've also seen that the installer complained about too small partitions. I had choosen a 100 GB partition but the installer complained that there is only 125 MB left. Finally I found out, that this happens if the partition is still mounted from a previous install attempt (or even from a manual mount).

igorpecovnik commented:
> We should never make it so easy for the user to delete their whole drive.

Added another check in the line.

> We should not use dd to wipe a whole drive just to remove partitions since this will unnecessarily wear down SD cards and SSDs which don't like to be written over and over again.

We only wipe our partition defs.

> We should not force the user to make a single big partition spanning 100% of their drive since with SSDs you somtimes want to use over-provisioning (meaning leaving like 10% free space unpartitioned on the SSD)

Agree, but this would require some work. Expanding this tool or ... introduce pick some solution like Calamares?

> We should not force the user to use ext4

We don't. Formating is done in next step. I have removed formating in the wipe part.

igorpecovnik commented:
>  I had choosen a 100 GB partition but the installer complained that there is only 125 MB left.

Yes, this part we need to look into. With my solution /  workaround this can't happen as we do auto partitioning. For most users this is just fine, but I do agree that we need some expansion of functionality. But before we do that, basic functions should work well.

igorpecovnik commented:
@ColorfulRhino can we proceed with this. Still better to have this in before developing something better.

hi-ko commented:
## short version

having something like `parted ${ARMBIAN_PARTITION} set 1 boot on` in arbian-install would have avoided my issue discussed in the forum.

## long version

I had three issues, described in the forum thread:

* inconsistent /etc/armbian-release file due to renaming BOARDFAMILY to rockchip64 which broke armbian-config / armbian-install
* unbootable system after running armbian-config to switch/update the kernel (which then may not fit to the bootloader config)
* unbootable system after running armbian-install from sdcard to install into a manual wiped partition due to missing boot flag on that partition

supporting multiple partitions should be always seen as best practice since it may be required to replace the armbian partition from time to time (since there is no such thing like a do-release-upgrade for now). Having one or more data partitions, which will guarantee to survive, could be a life saver then.

That does not necessarily mean, that all scenarios should be supported by the armbian-install. Experienced people have no problems in rsyning the sdcard to a custom partition or to wipe an already used partition but they may need to know the steps expected to be done and the knowledge how to configure/check the bootloader. at least for me u-boot is new territory. I will need to dig into u-boot and your scripts.

So my suggestion to start with is:

we should document (I could help on that)
- the steps to be done manually if *not* using armbian-install
- the _not_ supported scenarios by armbian-install

features I would like to have are:
* run armbian-install with parameters / without dialogue (e.g. `armbian-install --bootloader-mtd-flash /dev/nvme0n1p2 /mnt/armbian/boot/`)

igorpecovnik commented:
> features I would like to have are:

I would suggest to open a Jira ticket, so this information above is not lost. Not sure when this will be possible to implement, but chances are bigger if ticket exits. Here are basic instructions how-to: https://www.armbian.com/participate/

> we should document (I could help on that)

Add this into this secton:
https://docs.armbian.com/User-Guide_Getting-Started/#how-to-install-to-emmc-sata-nvme-usb
